src/components/Settings/Selection/context.ts

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import {SelectedSettingsPart, SettingsPage, getSelectedSettingsPart} from '../collect-settings';
+import {SettingsSelection} from './types';
+
+interface ContextValue extends SelectedSettingsPart {
+    selectedRef?: React.RefObject<HTMLDivElement>;
+}
+
+const defaultValue: ContextValue = {};
+
+const context = React.createContext(defaultValue);
```

Comment:
Why can't we use SettingsContext?

Reply:
We could, but I wanted to hide the selection logic from the user
If we put it in the `SettingsContext`, then the `selectedRef` and `SelectedSettingsPart` are going to be visible to the user through the `useSettingsContext` hook
Also the `SettingsContext` would get more complicated
That's why I chose to make a different, hidden context for selection